main.go

```diff
@@ -154,16 +172,74 @@ func main() {
 	}
 
 	go func() {
+
```

Comment:
how this should behave when we have more than a single replica ?
shouldn't it be invoked only by the leader ?

Reply:
No, not really, I mean, it'll fetch the config from the configmap or 3party secret provider.